code_sync/code_sync.py

```diff
@@ -2,39 +2,186 @@
 
 import argparse
 import os
+from pathlib import Path
 import subprocess
+import yaml
+from typing import Dict, Union
+
 
 cmd_str = 'watchmedo shell-command --recursive --patterns="{local_dir}*" --command="rsync --filter=\':- .gitignore\' ' \
           '--exclude \'*.ipynb\' --exclude \'.git\' --delete-after -rz --port {port} {local_dir} ' \
           '{target}:{remote_dir}" {local_dir}'
 
 epilog_str = '''
-Example for connecting to LeoMed:
-    code_sync --local_dir mylocaldir/ --remote_dir myremotedir/ --target medinfmk --port 2222\n
+EXAMPLE USAGE
+Register a project:
+    code_sync --register <project>
+
+code_sync a registered project:
+    code_sync <project>
+
+List all projects registered to code_sync:
+    code_sync --list
+
+Run code_sync with specific parameters:
+    code_sync --local_dir <mylocaldir/> --remote_dir <myremotedir/> --target <ssh_remote> --port 2222\n
 
 '''
 
+CONFIG_FILE_NAME = '.code_sync'
+
 
 def code_sync(local_dir, remote_dir, target, port=22):
     # clean up slashes
     local_dir = os.path.join(local_dir, '')
     remote_dir = os.path.join(remote_dir, '')
 
-    # subprocess.call()
+    print("Starting code_sync between {} and {}:{} ...".format(local_dir, target, remote_dir))
```

Comment:
Just stylistically: are you using format instead of f-strings because it need to be compatible with previous versions of Python?

Reply:
Good point! No, I just still forget sometimes. The current Python dependency is 3.6, which is where f-strings came into existence, so I'll updated all of them in the file (not just this instance).